gammapy/utils/docs.py

```diff
@@ -96,3 +100,81 @@ def gammapy_sphinx_ext_activate():
     # Register our directives and roles with Sphinx
     register_directive('gp-extra-image', ExtraImage)
     roles.register_local_role('gp-extra-notebook', notebook_role)
+
+def modif_nb_links(folder, url_docs):
+    """
+    Modifies links in raw and sphinx formatted notebooks and so they
+    point to and from the same version of the documentation. Adds a box to the
+    sphinx formatted notebooks with info and link to the ipynb file.
+    """
+
+    DOWNLOAD_CELL = """
+<div class='admonition note'>
+**This is a fixed-text formatted version of a Jupyter notebook.**
+
+You can download for each version of *gammapy* a
+[HTMLZip pack](http://readthedocs.org/projects/gammapy/downloads/) containing
+the whole HTML documentation and full collection of notebooks, so you can execute
+them in your local `_static/notebooks/` folder. You can also contribute with your
+own notebooks in this [GitHub repository](https://github.com/gammapy/gammapy-extra/tree/master/notebooks).
+
+**Source files:**
+[{nb_filename}](../_static/notebooks/{nb_filename}) |
+[{py_filename}](../_static/notebooks/{txt_filename})
+</div>"""
+
+    for filename in os.listdir(folder):
+        filepath = os.path.join(folder, filename)
+        if os.path.isfile(filepath) and filepath[-6:] == '.ipynb':
+            if folder=='notebooks':
+                py_filename = filename.replace('ipynb', 'py')
+                txt_filename = filename.replace('ipynb', 'txt')
+                ctx = dict(nb_filename=filename, py_filename=py_filename, txt_filename=txt_filename)
+                strcell = DOWNLOAD_CELL.format(**ctx)
+                nb = nbformat.read(filepath, as_version=nbformat.NO_CONVERT)
+                nb.cells.insert(0, new_markdown_cell(strcell))
+                nbformat.write(nb, filepath)
+            with open(filepath) as f:
+                txt = f.read()
+            if folder=='notebooks':
+                txt = re.sub(url_docs+'(.*?)html(\)|#)',r'..\1rst\2', txt, flags=re.M|re.I)
+            if folder=='_static/notebooks':
+                txt = re.sub(url_docs+'(.*?)html(\)|#)',r'..\/..\1html\2', txt, flags=re.M|re.I)
+            with open(filepath, "w") as f:
+                f.write(txt)
+
+def replace_extension(folder):
+    """
+    Replaces extension of .py files so they can be served for download
+    """
+
+    for filename in os.listdir(folder):
+        txtfilename = filename.replace('.py', '.txt')
+        os.rename(os.path.join(folder, filename), os.path.join(folder, txtfilename))
+
+def gammapy_sphinx_notebooks(setup_cfg):
+    """
+    Manages the processes for the building of sphinx formatted notebooks
+    """
+
+    url_docs = setup_cfg.get('url_docs')
+
+    # remove existing notebooks if rebuilding
+    if bool(setup_cfg.get('clean_notebooks')):
+        print('*** Cleaning notebooks')
```

Comment:
Maybe emit Sphinx log messages instead of just `print`?
http://www.sphinx-doc.org/en/stable/extdev/appapi.html#producing-messages-logging

Reply:
![captura de pantalla 2017-11-22 a las 9 53 11](https://user-images.githubusercontent.com/328261/33118084-fc3a28be-cf6a-11e7-9b6c-0da56f6c6e8f.png)


I'll use the logging API
http://www.sphinx-doc.org/en/stable/extdev/logging.html#logging-api

Reply:
@Bultako - Thanks for figuring out how to use logging with Sphinx!

Our docs / sphinx tooling will grow over time, so it's good if we do things properly.

While you're at it, would you mind also changing these lines further up in the file to log messages?
```
$ ack print gammapy/utils/docs.py 
        print('*** Found GAMMAPY_EXTRA = {}'.format(gammapy_extra_path))
        print('*** Nice!')
        print('*** gammapy-extra *not* found.')
        print('*** Set the GAMMAPY_EXTRA environment variable!')
        print('*** Docs build will be incomplete.')
        print('*** Notebook links will not be verified.')
```

Reply:
sure, it's done :)